kernel/kernel-api/src/main/java/io/delta/kernel/internal/IcebergCompatV2Utils.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (2023) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.delta.kernel.internal;
+
+import static java.lang.String.format;
+
+import io.delta.kernel.exceptions.KernelException;
+import io.delta.kernel.utils.DataFileStatus;
+
+/** Utility methods for validation and compatibility checks for Iceberg V2. */
+public class IcebergCompatV2Utils {
+  private IcebergCompatV2Utils() {}
+
+  /**
+   * Validate the given {@link DataFileStatus} that is being added as a {@code add} action to Delta
+   * Log. Currently, it checks that the statistics are not empty.
+   *
+   * @param dataFileStatus The {@link DataFileStatus} to validate.
+   */
+  public static void validDataFileStatus(DataFileStatus dataFileStatus) {
+    if (!dataFileStatus.getStatistics().isPresent()) {
```

Comment:
This is sufficient since DataFileStatistics always has non-null `numRecords` right? Maybe just add a clarifying comment here since that's not obvious